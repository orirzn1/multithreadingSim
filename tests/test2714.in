L9
S8
O13
N12

T0
I@0x00000000
STORE $7, $6, $3
ADD $1, $1, $2
SUBI $4, $5, 0x3
LOAD $7, $3, 0x1
ADD $1, $4, $7
STORE $4, $1, 0x4
SUB $4, $7, $4
ADDI $2, $0, 0x1
STORE $4, $6, 0x3
LOAD $3, $2, $7
HALT $0

T1
I@0x00000000
LOAD $5, $0, 0x4
ADDI $3, $4, 0x1
STORE $6, $6, $1
SUBI $7, $5, 0x3
SUBI $4, $2, 0x1
STORE $5, $7, $2
LOAD $0, $0, 0x3
STORE $2, $0, 0x0
ADD $7, $3, $7
SUBI $6, $6, 0x4
HALT $0

T2
I@0x00000000
ADD $5, $5, $4
SUBI $6, $4, 0x1
SUBI $7, $4, 0x2
LOAD $6, $0, $2
LOAD $4, $3, $3
SUBI $0, $6, 0x3
ADDI $6, $2, 0x4
HALT $0

T3
I@0x00000000
ADDI $1, $7, 0x2
SUB $3, $3, $1
ADDI $1, $4, 0x4
LOAD $0, $0, $5
ADD $5, $2, $7
ADDI $1, $6, 0x4
HALT $0

T4
I@0x00000000
LOAD $6, $6, 0x3
SUB $2, $5, $5
ADD $7, $7, $4
LOAD $3, $6, 0x3
SUB $0, $0, $6
ADD $7, $5, $1
ADDI $7, $0, 0x1
ADDI $2, $2, 0x4
HALT $0

T5
I@0x00000000
LOAD $3, $2, 0x0
ADDI $5, $4, 0x2
STORE $2, $5, 0x0
ADD $2, $6, $1
ADD $2, $7, $5
HALT $0

T6
I@0x00000000
STORE $7, $5, 0x3
ADD $5, $6, $7
ADDI $7, $1, 0x4
SUBI $6, $6, 0x2
HALT $0

T7
I@0x00000000
LOAD $7, $1, 0x4
ADD $5, $7, $6
ADDI $5, $5, 0x2
SUBI $2, $3, 0x2
SUB $2, $5, $1
ADD $5, $7, $2
ADD $0, $6, $6
SUBI $3, $7, 0x1
HALT $0

T8
I@0x00000000
LOAD $3, $4, $7
SUBI $0, $2, 0x1
SUBI $2, $5, 0x4
LOAD $4, $1, 0x4
SUB $5, $4, $0
SUBI $3, $4, 0x2
ADD $5, $2, $0
SUB $4, $3, $6
HALT $0

T9
I@0x00000000
LOAD $4, $6, 0x2
ADD $7, $0, $5
ADD $4, $1, $2
SUBI $4, $2, 0x4
ADDI $2, $0, 0x4
ADDI $1, $5, 0x1
ADD $7, $3, $3
ADD $4, $3, $1
STORE $4, $1, $0
HALT $0

T10
I@0x00000000
STORE $1, $7, $1
SUB $4, $1, $4
STORE $0, $0, $6
SUB $3, $4, $2
HALT $0

T11
I@0x00000000
STORE $2, $2, 0x4
ADD $2, $5, $6
STORE $3, $4, 0x2
ADDI $5, $7, 0x2
LOAD $4, $1, 0x2
HALT $0

D@0x0
0x5
0xA
0xA
0x2
0xA
0x4
