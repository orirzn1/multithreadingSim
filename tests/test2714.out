
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0x0	R1 = 0xFFFFFFFD	R2 = 0x1	R3 = 0xFFFFFFFD	R4 = 0x3	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0xFFFFFFFD	R1 = 0x0	R2 = 0x0	R3 = 0x1	R4 = 0xFFFFFFFF	R5 = 0xA	R6 = 0xFFFFFFFC	R7 = 0x8
Register file thread id 2:
	R0 = 0xFFFFFFFD	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x4	R7 = 0xFFFFFFFE
Register file thread id 3:
	R0 = 0x0	R1 = 0x4	R2 = 0x0	R3 = 0xFFFFFFFE	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0x4	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x1
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x2	R3 = 0x0	R4 = 0x0	R5 = 0x2	R6 = 0x0	R7 = 0x0
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFE	R7 = 0x4
Register file thread id 7:
	R0 = 0x0	R1 = 0x0	R2 = 0xC	R3 = 0x9	R4 = 0x0	R5 = 0x16	R6 = 0x0	R7 = 0xA
Register file thread id 8:
	R0 = 0xFFFFFFFF	R1 = 0x0	R2 = 0xFFFFFFFC	R3 = 0xFFFFFFFE	R4 = 0xFFFFFFFE	R5 = 0xFFFFFFFB	R6 = 0x0	R7 = 0x0
Register file thread id 9:
	R0 = 0x0	R1 = 0x1	R2 = 0x4	R3 = 0x0	R4 = 0x1	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 10:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 11:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFD	R5 = 0x2	R6 = 0x0	R7 = 0x0
Blocked MT CPI for this program 6.281250

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x0	R1 = 0xFFFFFFFD	R2 = 0x1	R3 = 0x1	R4 = 0x3	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0xFFFFFFFD	R1 = 0x0	R2 = 0x0	R3 = 0x1	R4 = 0xFFFFFFFF	R5 = 0x0	R6 = 0xFFFFFFFC	R7 = 0xFFFFFFFE
Register file thread id 2:
	R0 = 0xFFFFFFFD	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x4	R7 = 0xFFFFFFFE
Register file thread id 3:
	R0 = 0x0	R1 = 0x4	R2 = 0x0	R3 = 0xFFFFFFFE	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0x4	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x1
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x2	R3 = 0x0	R4 = 0x0	R5 = 0x2	R6 = 0x0	R7 = 0x0
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFE	R7 = 0x4
Register file thread id 7:
	R0 = 0x0	R1 = 0x0	R2 = 0x2	R3 = 0xFFFFFFFF	R4 = 0x0	R5 = 0x2	R6 = 0x0	R7 = 0x0
Register file thread id 8:
	R0 = 0xFFFFFFFF	R1 = 0x0	R2 = 0xFFFFFFFC	R3 = 0xFFFFFFFE	R4 = 0xFFFFFFFE	R5 = 0xFFFFFFFB	R6 = 0x0	R7 = 0x0
Register file thread id 9:
	R0 = 0x0	R1 = 0x1	R2 = 0x4	R3 = 0x0	R4 = 0x1	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 10:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 11:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x2	R6 = 0x0	R7 = 0x0
Finegrained Multithreading CPI for this program 1.083333

